core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeInMemorySorter.java

```diff
@@ -301,6 +323,19 @@ public SortedIterator getSortedIterator() {
       }
     }
     totalSortTimeNanos += System.nanoTime() - start;
-    return new SortedIterator(pos / 2, offset);
+    if (nullPos > 0) {
+      assert radixSortSupport != null : "Nulls are only stored separately with radix sort";
+      LinkedList<UnsafeSorterIterator> queue = new LinkedList<>();
+      if (radixSortSupport.sortDescending()) {
+        queue.add(new SortedIterator((pos - nullPos) / 2, offset));
```

Comment:
nit: can we add a comment along the lines of `Nulls are smaller than non-nulls...`?